src/transformStateWithClones.js

```diff
@@ -8,6 +8,27 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const copyState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copyState, action.extraData);
+        break;
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => delete copyState[key]);
+        break;
+      case 'clear':
+        for (const key in copyState) {
+          delete copyState[key];
+        }
+        break;
+    }
```

Comment:
The switch statement is missing a default case. It's important to handle unexpected `action.type` values gracefully, possibly by throwing an error or ignoring the action.